stream_alert_cli/test.py

```diff
@@ -68,98 +71,223 @@ def __init__(self, context, print_output):
                 Warnings and errors captrued during rule processor testing
                 will still be written to stdout regardless of this setting.
         """
+        # Create the RuleProcessor. Passing a mocked context object with fake
+        # values and False for suppressing sending of alerts to alert processor
+        self.processor = StreamAlert(context, False)
+        # Use a list of status_messages to store pass/fail/warning info
+        self.status_messages = []
+        self.total_tests = 0
         self.all_tests_passed = True
-        self.context = context
-        # Create the topic used for the mocking of alert sending
-        # This is used in stream_alert/rule_processor/sink.py to 'send' alerts
-        sns_client = boto3.client('sns', region_name='us-east-1')
-        sns_client.create_topic(Name='test_streamalerts')
-        # Create a list for pass/fails. The first value in the list is a
-        # list of tuples for failures, and the second is list of tuples for
-        # passes. Tuple is (rule_name, rule_description)
-        self.rules_fail_pass_warn = ([], [], [])
         self.print_output = print_output
-        self.invalid_log_messages = []
 
-    def test_processor(self, rules):
+    def test_processor(self, filter_rules, validate_only=False):
         """Perform integration tests for the 'rule' Lambda function
 
         Args:
-            rules (list|None): Specific rule names (or None) to restrict
+            filter_rules (list|None): Specific rule names (or None) to restrict
                 testing to. This is passed in from the CLI using the --rules option.
+            validate_only [bool=False]: If true, validation of test records will occur
```

Comment:
Nit: should be `validate_only (bool)`. Types should be in parens (not brackets), and the default value need not be specified since it's in the function definition